src/esp/CMakeLists.txt

```diff
@@ -17,15 +17,595 @@ if(BUILD_WARNINGS_AS_ERRORS)
-add_subdirectory(physics)
-add_subdirectory(nav)
-add_subdirectory(sensor)
-add_subdirectory(sim)
+##############################
+
+set(habsim_SOURCES ${core_SOURCES})
+list(
+  APPEND
+  habsim_SOURCES
+  ${geo_SOURCES}
+  ${gfx_SOURCES}
+  ${assets_SOURCES}
+  ${metadata_SOURCES}
+  ${io_SOURCES}
+  ${scene_SOURCES}
+  ${physics_SOURCES}
+  ${nav_SOURCES}
+  ${sensor_SOURCES}
+  ${sim_SOURCES}
+)
+
+add_library(
+  habsim STATIC
+  ${habsim_SOURCES}
+)
+
+target_include_directories(
+  habsim PUBLIC ${PROJECT_BINARY_DIR}
+  PRIVATE "${DEPS_DIR}/tinyxml2" "${DEPS_DIR}/recastnavigation/Detour/Include"
+          "${DEPS_DIR}/recastnavigation/Recast/Include"
+)
+
+if(BUILD_WITH_CUDA)
+  add_library(noise_model_kernels STATIC ${noise_model_SOURCES})
+  target_link_libraries(noise_model_kernels PUBLIC ${CUDART_LIBRARY})
+  target_include_directories(
+    noise_model_kernels PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+  )
+  target_compile_features(noise_model_kernels PUBLIC cxx_std_11)
+
+  target_link_libraries(habsim PUBLIC noise_model_kernels ${CUDART_LIBRARY})
+
+  target_include_directories(habsim PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+endif()
+
+if(BUILD_WITH_AUDIO)
+  include_directories(${DEPS_DIR}/rlr-audio-propagation/RLRAudioPropagationPkg/headers)
+  target_link_libraries(habsim PUBLIC ${RLRAudioPropagation_LIBRARY})
+endif()
+
+target_link_libraries(
+  habsim
+  PUBLIC gfx_batch
+         Magnum::AnyImageImporter
+         Magnum::AnySceneImporter
+         Magnum::GL
+         Magnum::Magnum
+         Magnum::MaterialTools
+         Magnum::MeshTools
+         Magnum::DebugTools
+         Magnum::SceneGraph
+         Magnum::SceneTools
+         Magnum::Shaders
+         Magnum::Trade
+         Magnum::Primitives
+         MagnumPlugins::BasisImporter
+         MagnumPlugins::GltfImporter
+         MagnumPlugins::PrimitiveImporter
+         MagnumPlugins::StanfordImporter
+         MagnumPlugins::StbImageImporter
+         MagnumPlugins::StbImageConverter
+         MagnumIntegration::Eigen
+         Corrade::Utility
+         Magnum::AnyImageConverter
+  PRIVATE tinyxml2 Detour Recast
+)
+
+# Link windowed application library if needed
+if(BUILD_GUI_VIEWERS)
+  target_link_libraries(habsim PUBLIC Magnum::GlfwApplication)
+endif()
+
+if(BUILD_WITH_BACKGROUND_RENDERER)
+  target_link_libraries(habsim PUBLIC atomic_wait)
+endif()
+
+if(BUILD_ASSIMP_SUPPORT)
+  target_link_libraries(
+    habsim PUBLIC MagnumPlugins::AssimpImporter PRIVATE Assimp::Assimp
+  )
+endif()
+
+if(OpenMP_CXX_FOUND)
+  target_link_libraries(habsim PUBLIC OpenMP::OpenMP_CXX)
+endif()
+
+target_link_libraries(
+  habsim
+  PUBLIC Magnum::WindowlessApplication
+)
+if(BUILD_WITH_BULLET)
+  target_link_libraries(habsim PUBLIC MagnumIntegration::Bullet Bullet::Dynamics)
+endif() #BUILD_WITH_BULLET
+
+# enable when finished? Slows compilation substantially
+# set_directory_properties(PROPERTIES CORRADE_USE_PEDANTIC_FLAGS ON)
```

Comment:
That fix we talked about is in https://github.com/mosra/corrade/commit/280e6b6874e0902086e321d733ce0e90e0c52920. But again, there's _quite a few_ breaking changes in Magnum since the last update, so I'm not sure when it would be the best time to update here.

Reply:
We can discuss the breaking changes in slack. TBH I would think it worthwhile to investigate updating Habitat to the most recent Magnum so any future Magnum growth will be more easily incorporated should it be useful, but of course that would depend on the scope of the changes.

Reply:
What I wanted to say is that I unfortunately don't currently have the bandwidth to adapt to the changes. In C++ the changed APIs are just emitting deprecation warnings, so that's fine (you can upgrade gradually as necessary, for example), but for Python there's no such deprecation workflow in place, and so all the code needs to be adapted immediately. 

For example there are no application mouse events anymore, it's a [PointerEvent](https://doc.magnum.graphics/python/magnum/platform/glfw/Application/PointerEvent/) now, to have a single abstraction for a mouse, pen and touch. Again, in C++ the old [MouseEvent](https://doc.magnum.graphics/magnum/classMagnum_1_1Platform_1_1GlfwApplication_1_1MouseEvent.html) interface is deprecated, but in Python it's just gone.

If you'd have time to do these changes, however, that's a way... :)